src/main/java/picard/sam/SamToFastqWithTags.java

```diff
@@ -0,0 +1,213 @@
+
+@CommandLineProgramProperties(
+        summary = SamToFastqWithTags.USAGE_SUMMARY + SamToFastqWithTags.USAGE_DETAILS,
+        oneLineSummary = SamToFastqWithTags.USAGE_SUMMARY,
+        programGroup = ReadDataManipulationProgramGroup.class)
+public class SamToFastqWithTags extends SamToFastq {
+    static final String USAGE_SUMMARY = "Converts a SAM or BAM file to FASTQ.";
+    static final String USAGE_DETAILS = "This will create the fastq from read sequences alongside two fastq files whose sequence is generated from tags.  " +
+            "One will be converted from the \"CR\" tag with quality from the \"CY\" tag.  The other fastq will be " +
+            "converted from the \"CB\" and \"UR\" tags concatenated together with no separator (not specified on " +
+            "command line) with the qualities coming from the \"CY\" and \"UY\" tags concatenated together.  The two files " +
+            "will be named CR.fastq and CB_UR.fastq" +
+            "<br />" +
+            "<pre>" +
+            "java -jar picard.jar SamToFastqWithTags <br />" +
+            "     I=input.bam<br />" +
+            "     FASTQ=output.fastq<br />" +
+            "     SEQUENCE_TAG_GROUP=CR<br />" +
+            "     QUALITY_TAG_GROUP=CY<br />" +
+            "     SEQUENCE_TAG_GROUP=\"CB,UR\"<br />" +
+            "     QUALITY_TAG_GROUP=\"CY,UY\"" +
+            "</pre>" +
+            "<hr />";
+
+    @Argument(shortName = "STG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read sequence", minElements = 1)
+    public List<String> SEQUENCE_TAG_GROUP;
+
+    @Argument(shortName = "QTG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read qualities", optional = true)
+    public List<String> QUALITY_TAG_GROUP;
+
+    @Argument(shortName = "SEP", doc = "List of sequences to put in between each comma separated list of sequence tags in each SEQUENCE_TAG_GROUP (STG)", optional = true)
+    public List<String> TAG_GROUP_SEPERATOR;
+
+    @Argument(shortName = "GZOPTG", doc = "Compress output FASTQ files per Tag grouping using gzip and append a .gz extension to the file names.")
+    public Boolean COMPRESS_OUTPUTS_PER_TAG_GROUP = false;
+
+    private final Log log = Log.getInstance(SamToFastqWithTags.class);
+
+    private final static String TAG_SPLIT_DEFAULT_SEP = "";
+    private final static String TAG_SPLIT_DEFAULT_QUAL = "~";
+
+    private ArrayList<String[]> SPLIT_SEQUENCE_TAGS;
+    private ArrayList<String[]> SPLIT_QUALITY_TAGS;
+    private ArrayList<String> SPLIT_SEPARATOR_TAGS;
+
+    @Override
+    protected void initializeAdditionalWriters() {
+        setupTagSplitValues();
+    }
+
+    @Override
+    protected void handleAdditionalRecords(SAMRecord currentRecord, Map<SAMReadGroupRecord, List<FastqWriter>> tagWriters, SAMRecord read1, SAMRecord read2) {
+        if (currentRecord.isSecondaryOrSupplementary() && !INCLUDE_NON_PRIMARY_ALIGNMENTS)
```

Comment:
please use `{` `}` around the `if` body

Reply:
Is someone going to go through the codebase and fix these?  Its hard to tell what is and isnt proper style for this repo

Reply:
I think the desired target is the Sun/Oracle Java coding standard (Google also has a Java coding standard which also requires braces around if/else/etc bodies). Regarding fixing all the code, I don't know if it's worth the effort to reformat everything, personally I would reformat code as it's modified due to other work being done.